app/Controllers/Http/SearchController.ts

```diff
@@ -0,0 +1,47 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Definition from 'App/Models/Definition'
+
+export default class SearchController {
+  public async index({ request, response }: HttpContextContract) {
+    const { char, term } = request.qs()
+    const formattedTerm = decodeURI(term).trim()
+
+    if (!char || !formattedTerm) {
+      return response.status(400).send({
+        code: 400,
+        status: 'Bad Request',
+        message: 'Term or char are required',
+      })
+    }
+
+    try {
+      const QUERY_CHAR = await Definition.query().where('term', 'like', `${char}%`).distinct('term')
+      const QUERY_TERM = await Definition.query()
+        .where('term', 'like', `%${formattedTerm}%`)
+        .distinct('term')
+
+      const terms = char ? QUERY_CHAR : QUERY_TERM
```

Comment:
don't make a double request to database, you can customize the query based on the conditions that have been created